mapgen.lua

```diff
@@ -468,7 +523,8 @@ function vmg.generate(minp, maxp, seed)
 								data[ivm] = c_stone
 							end
 						end
-					elseif v11 + v12 > 2 ^ (y / lava_depth) and y <= lava_max_height then
+						-- Allow lava above ground to max_height, but see nodes.lua.
+					elseif y <= lava_max_height and v11 + v12 > 2^((y - lava_max_height) / lava_depth) then
```

Comment:
With lava_max_height = 31000 to enable surface lava, most of the caves are completely filled by lava. This could be fixed by `y / lava_depth` instead of `(y - lava_max_height) / lava_depth`